controllers/routemonitor/routemonitor_controller_supplement_test.go

```diff
@@ -101,72 +141,483 @@ var _ = Describe("Routemonitor", func() {
+	// 			// Assert
+	// 			Expect(err).To(HaveOccurred())
+	// 			Expect(err).To(MatchError(customError))
+	// 		})
+	// 	})
+
+	// 	When("'Get' return an 'NotFound' error", func() {
+	// 		// Arrange
+	// 		BeforeEach(func() {
+	// 			getErrorResponse = notFoundErr
+	// 		})
+	// 		It("should do nothing", func() {
+	// 			// Act
+	// 			err := routeMonitorReconciler.DeleteBlackBoxExporterService(ctx)
+	// 			// Assert
+	// 			Expect(err).NotTo(HaveOccurred())
+	// 		})
+	// 	})
+
+	// 	When("'Delete' return an an  error", func() {
+	// 		// Arrange
+	// 		BeforeEach(func() {
+	// 			deleteCalledTimes = 1
+	// 			deleteErrorResponse = customError
+	// 		})
+	// 		It("should do nothing", func() {
+	// 			// Act
+	// 			err := routeMonitorReconciler.DeleteBlackBoxExporterService(ctx)
+	// 			// Assert
+	// 			Expect(err).To(HaveOccurred())
+	// 			Expect(err).To(MatchError(customError))
+	// 		})
+	// 	})
+
+	// 	When("'Delete' succeeds", func() {
+	// 		// Arrange
+	// 		BeforeEach(func() {
+	// 			deleteCalledTimes = 1
+	// 		})
+	// 		It("should succeed", func() {
+	// 			// Act
+	// 			err := routeMonitorReconciler.DeleteBlackBoxExporterService(ctx)
+	// 			// Assert
+	// 			Expect(err).NotTo(HaveOccurred())
+	// 		})
+	// 	})
+
+	// })
+
+	// Describe("DeleteBlackBoxExporterDeployment", func() {
+	// 	BeforeEach(func() {
+	// 		getCalledTimes = 1
+	// 		routeMonitorReconcilerClient = mockClient
+	// 	})
+
+	// 	When("'Get' return an error", func() {
+	// 		// Arrange
+	// 		BeforeEach(func() {
+	// 			getErrorResponse = customError
+	// 		})
+	// 		It("should bubble the error up", func() {
+	// 			// Act
+	// 			err := routeMonitorReconciler.DeleteBlackBoxExporterDeployment(ctx)
+	// 			// Assert
+	// 			Expect(err).To(HaveOccurred())
+	// 			Expect(err).To(MatchError(customError))
+	// 		})
+	// 	})
+
+	// 	When("'Get' return an 'NotFound' error", func() {
+	// 		// Arrange
+	// 		BeforeEach(func() {
+	// 			getErrorResponse = notFoundErr
+	// 		})
+	// 		It("should do nothing", func() {
+	// 			// Act
+	// 			err := routeMonitorReconciler.DeleteBlackBoxExporterDeployment(ctx)
+	// 			// Assert
+	// 			Expect(err).NotTo(HaveOccurred())
+	// 		})
+	// 	})
+
+	// 	When("'Delete' return an an  error", func() {
+	// 		// Arrange
+	// 		BeforeEach(func() {
+	// 			deleteCalledTimes = 1
+	// 			deleteErrorResponse = customError
+	// 		})
+	// 		It("should do nothing", func() {
+	// 			// Act
+	// 			err := routeMonitorReconciler.DeleteBlackBoxExporterDeployment(ctx)
+	// 			// Assert
+	// 			Expect(err).To(HaveOccurred())
+	// 			Expect(err).To(MatchError(customError))
+	// 		})
+	// 	})
+
+	// 	When("'Delete' succeeds", func() {
+	// 		// Arrange
+	// 		BeforeEach(func() {
+	// 			deleteCalledTimes = 1
+	// 		})
+	// 		It("should succeed", func() {
+	// 			// Act
+	// 			err := routeMonitorReconciler.DeleteBlackBoxExporterDeployment(ctx)
+	// 			// Assert
+	// 			Expect(err).NotTo(HaveOccurred())
+	// 		})
+	// 	})
+
+	// })
```

Comment:
If this code is not needed anymore, I recommend removing it.

Reply:
it is needed, but writing the tests became more cumbersome. (there is  not enough coverage and all these fucntions are not exposed anymore